storage-node-v2/src/services/webApi/controllers/publicApi.ts

```diff
@@ -296,17 +301,59 @@ async function validateTokenRequest(api: ApiPromise, tokenRequest: UploadTokenRe
 /**
  * Validates file size. It throws an error when file size exceeds the limit
  *
- * @param fileSize - runtime API promise
+ * @param api - runtime API promise
+ * @param fileSize - file size to validate
  * @returns void promise.
  */
-function verifyFileSize(fileSize: number) {
-  const MAX_FILE_SIZE = 1000000 // TODO: Get this const from the runtime
+async function verifyFileSize(api: ApiPromise, fileSize: number) {
+  const maxRuntimeFileSize = await api.consts.storage.maxDataObjectSize.toNumber()
 
-  if (fileSize > MAX_FILE_SIZE) {
+  if (fileSize > maxRuntimeFileSize) {
     throw new WebApiError('Max file size exceeded.', 400)
   }
 }
 
+/**
+ * Validates the runtime info for the data object. It verifies contentID,
+ * file size, and 'accepted' status.
+ *
+ * @param api - runtime API promise
+ * @param bagId - bag ID
+ * @param dataObjectId - data object ID to validate in runtime
+ * @param fileSize - file size to validate
+ * @param hash - file multihash
+ * @returns void promise.
+ */
+async function verifyDataObjectInfo(
+  api: ApiPromise,
+  bagId: BagId,
+  dataObjectId: number,
+  fileSize: number,
+  hash: string
+) {
+  const dataObject = await api.query.storage.dataObjectsById(bagId, dataObjectId)
+
+  if (dataObject.accepted.valueOf()) {
+    throw new WebApiError(`Data object had been already accepted ID = ${dataObjectId}`, 400)
+  }
```

Comment:
We haven't decided on the 'allowing already accepted objects' issue yet, and I didn't add the 'data object accepted' check on the runtime side. I was planning to add the 'storage node side' check as a fail-fast solution. 
Regarding possible attack vectors from the storage providers - transaction fees and slashing are our current mitigations.

However, I like your idea because we are still thinking about re-uploading data objects on the failure of the single holder (we also need to think about the DoS from the client of the storage node - multiple uploading of the same large file if we allow re-uploading the data object).

Anyway, It's easy to make it strict again (possibly during the auth-token creation to counter mentioned DoS).